src/Shared/Microsoft.DotNet.GitHub.Authentication/InstallationLookup.cs

```diff
@@ -9,7 +9,9 @@ namespace Microsoft.DotNet.GitHub.Authentication
     public interface IInstallationLookup
     {
         Task<long> GetInstallationId(string repositoryUrl);
+        Task<long> GetInstallationIdForApp(string appName, string repositoryUrl);
```

Comment:
I feel like I'd personally prefer a factory factory or something...  Normal usage would just take the IInstallationLookup.  But if you needed to get one for a specific app, that really feels like something that should be on a different type, that, most of the time, people don't need to worry about.

So you'd have `mulitpleInstallationHandler.GetInstallationLookup(appName).GetInstallationid()` or something like that, let the "IInstallationLookup" stay the way that it is, and let the fact that _sometimes_ it has a dependency on an app name be something elses problem.  More likely a given class is only going to care about a single installation, so it would just have in it's constructor `_installation = multiple.GetInstallationLookup(appName)`, and the rest of the code doesn't need to worry about it.